doc/resolver_dev.md

```diff
@@ -1,20 +1,172 @@
+
+*Inflection Request Sequence*
+
+1. The first 4 characters (i.e. "ark:" or "doi:") are removed from `{request_identifier}` yielding `{request_value}`.
+2. 
+
+the following parts are parsed:
+```
+{request_identifier} = {scheme} + ":" + {value} [+ {suffix}] [+ "?info" | "??"]
+```
+
+where:
+
+`scheme` is one of `ark` or `doi`<br />
+`value` is an identifier value in EZID<br />
+`suffix` 0..n characters after the `value`<br />
+`?info` or `??` signifies an inflection request<br />
+
+Resolution proceeds in the following manner:
+
+1. `request_identifier` is split to `{scheme}`, `{identifier_value}`, and `{scheme}` is converted to lowercase.
+2. If `{request_identifier}` ends with `?info` or `??` the 
+3. If `scheme` is `"doi"` a 302 redirect with location `"https://doi.org/{value}/{suffix}"` is returned. 
+4. `request_identifier` is normalized to `normalized_identifier_request` according the rules of `scheme`
+5. If `normalized_identifier_request` does not appear to be a valid identifier, a `404 Not Found` response is returned.
+6. A prefix search is performed to determine the `registered_identifier` that matches the start of `normalized_identifier_request`
+7. If a `registered_identifier` is not found, a `404 Not Found` error is returned.
+8. If `registered_identifier` has `RESERVED` status, a `404 Not Found` error i s returned.
+9. If `registered_identifier` has an `UNAVAILABLE` status, a HTTP 302 Temporary Redirect is returned with `Location` being the tombstone page for the identifier.
+10. `suffix` is the string remaining after subtracting `identifier` from the start of the string. This may include a query portion, i.e. key-value pairs.
+11. `Location` is constructed as `identifier.target + suffix`
+12. A HTTP `302 Temporary Redirect` response is returned, with `Location`. A response body of `JSON` is included, and the `Last-Modified` response header is set to the time that the record was last updated.
+
+
+
+
+
+
+If the identifier scheme is "doi", the client is redirected to:
+
+```
+https://dx.doi.net/{value}
+```
+
+If the identifier scheme is "ark":
+
+```
+{value} = ["/"] + {NAAN} + ["/" + {suffix}]
+```
+
+If the value is not an EZID shoulder, the client is redirected to:
+
+```
+https://n2t.net/{identifier}
+```
+
+If the value is a shoulder managed by EZID:
+
+If the status is "R" ("Reserved"), a status code of 403 Forbidden is returned.
+
+If the status is "U" ("Unavailable"), a status code of 308 Permanent Redirect is returned with a Location header containing the tombstone page for that identifier. The tombstone page will typically be:
```

Comment:
Is this 302?

Reply:
resolver_dev.md contains notes generated during the resolver service development. The document does not reflect the final state of implementation, instead please refer to the API documentation and source code for actual implementation details.